samples/debug/memfault/src/main.c

```diff
@@ -68,33 +83,57 @@ static void button_handler(uint32_t button_states, uint32_t has_changed)
 		ARG_UNUSED(i);
 	} else if (has_changed & DK_BTN3_MSK) {
 		/* DK_BTN3_MSK is Switch 1 on nRF9160 DK. */
-		int err = memfault_metrics_heartbeat_add(
-			MEMFAULT_METRICS_KEY(Switch1ToggleCount), 1);
-
+		int err =
+			memfault_metrics_heartbeat_add(MEMFAULT_METRICS_KEY(Switch1ToggleCount), 1);
 		if (err) {
 			LOG_ERR("Failed to increment Switch1ToggleCount");
 		} else {
 			LOG_INF("Switch1ToggleCount incremented");
 		}
 	} else if (has_changed & DK_BTN4_MSK) {
 		/* DK_BTN4_MSK is Switch 2 on nRF9160 DK. */
-		MEMFAULT_TRACE_EVENT_WITH_LOG(Switch2Toggled,
-					      "Switch state: %d",
+		MEMFAULT_TRACE_EVENT_WITH_LOG(Switch2Toggled, "Switch state: %d",
 					      buttons_pressed & DK_BTN4_MSK ? 1 : 0);
 		LOG_INF("Switch2Toggled event has been traced, button state: %d",
 			buttons_pressed & DK_BTN4_MSK ? 1 : 0);
 	}
 }
+static void on_net_event_l4_connected(void)
+{
+	network_connected();
+}
+
+static void on_net_event_l4_disconnected(void)
+{
+	network_disconnected();
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb, uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_L4_CONNECTED:
+		LOG_INF("Network connectivity established");
+		on_net_event_l4_connected();
+		break;
+	case NET_EVENT_L4_DISCONNECTED:
+		LOG_INF("Network connectivity lost");
+		on_net_event_l4_disconnected();
+		break;
+	default:
+		LOG_DBG("Unknown event: 0x%08X", event);
+		return;
+	}
+}
 
 static void on_connect(void)
 {
 #if IS_ENABLED(MEMFAULT_NCS_LTE_METRICS)
 	uint32_t time_to_lte_connection;
-
 	/* Retrieve the LTE time to connect metric. */
-	memfault_metrics_heartbeat_timer_read(
-		MEMFAULT_METRICS_KEY(Ncs_LteTimeToConnect),
-		&time_to_lte_connection);
+	memfault_mek_sem_take(&nw_connected_sem, K_FOREVER);
```

Comment:
I think there's a typo in this line which will cause a compilation fail

Reply:
seems like I accidentally managed to add a k_sem_take in the middle of the function call, not sure when and how that happend, but it hasn't seem to affect the building of the sample.